### ISSUE TITLE
Cost reducer vs Hybrid Mana Symbols

### ISSUE THREAD

**Issue by kevlahnota** [WagicProject/wagic#735]
I think we have missing cost reduction on hybrid mana symbols.
Example you have 2 Rangemonger in play and have Boros Reckoner in Hand, if we apply cost reduction automatically, we can pay {RW} (either W or R, since we can't choose what half we will pay for the other two mana symbol that become {0W}{0W} when the cost reducer is applied, we choose 0 automatically) , so the cost we must pay will become {RW}.
![Ragemonger](http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=378525&type=card)![Boros Reckoner](http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=366385&type=card)

**Comment by zethfoxster** [comment]
thats tricky, got to make sure we can still pay cost like {w1}{rg}{b}

**Comment by kevlahnota** [comment]
Just got an idea how to get excess cost to reduce(inside compute new cost before we apply the reduced cost)


```

    ManaCost * excess = NEW ManaCost(); //declared above


        if (card->getIncreasedManaCost()->getConvertedCost())
            original->add(card->getIncreasedManaCost());
        //before removing get the diff for excess
        if(card->getReducedManaCost()->getConvertedCost())
        {
            for(int xc = 0; xc < 7;xc++)
            {//if the diff is more than 0
                if(card->getReducedManaCost()->Diff(original)->getCost(xc))
                {
                    int count = card->getReducedManaCost()->Diff(original)->getCost(xc);
                    excess->add(xc,count);
                }
            }
        }
        //apply reduced
        if (card->getReducedManaCost()->getConvertedCost())
            original->remove(card->getReducedManaCost());

      //try to reduce hybrid
        if (excess->getConvertedCost())
        {
            //function for hybrid replacing;
        }
```

I hope this is possible...

**Comment by kevlahnota** [comment]
fixed #744
